### ISSUE TITLE
Unable to SIGTERM systembridgebackend when run with GUI

### ISSUE THREAD

**Issue by ScottG489** [timmo001/system-bridge#1888]
### Description

I'm unable to stop `python3 -m systembridgebackend` by sending it a `SIGTERM` and must kill it and the GUI process (`python3 -m systembridgegui main`) manually with `SIGKILL`.

It seems like maybe it's spawning another process that isn't a child of the original?
```sh
ps uf -A | grep systembridge
scott    1420631  8.4  0.4 923244 105872 pts/35  Sl+  10:03   0:01  |   \_ python3 -m systembridgebackend
scott    1420751  8.2  0.7 1200888 185444 pts/35 Sl+  10:03   0:00  |       \_ /usr/bin/python3 -m systembridgegui main
scott    1420995  0.0  0.0   6528   920 pts/36   S+   10:04   0:00  |   \_ grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox systembridge
scott    1405237  0.1  0.7 1188576 185500 pts/35 Sl   09:56   0:00 /usr/bin/python3 -m systembridgegui main
```

As you can see there are two instances of `systembridgegui`

### What Platform / OS are you running?

Linux

### What version are you running?

3.4.3

### Anything in the logs or a references that might be useful?

Not sure if these logs are useful at all. There are some (likely unrelated) SQL errors which have already been reported in a separate issue. You can see at the end where it's killed (`^C`) and says it's stopped but continues to hang:
```
2022-07-27 09:56:00 INFO (MainThread) [__main__.Main] System Bridge 3.4.3: Startup
2022-07-27 09:56:00 INFO (MainThread) [__main__.Main] Autostart enabled: False
2022-07-27 09:56:00 INFO (MainThread) [systembridgebackend.server.Server] Serving frontend from: /home/scott/.local/lib/python3.9/site-packages/systembridgefrontend/out
2022-07-27 09:56:00 INFO (MainThread) [systembridgebackend.server.Server] Starting server on port: 9170
[2022-07-27 09:56:00 -0400] [1405161] [WARNING] Sanic is running in PRODUCTION mode. Consider using '--debug' or '--dev' while actively developing your application.
2022-07-27 09:56:00 WARNING (MainThread) [sanic.error] Sanic is running in PRODUCTION mode. Consider using '--debug' or '--dev' while actively developing your application.
[2022-07-27 09:56:01 -0400] [1405161] [INFO] Sanic Extensions:
2022-07-27 09:56:01 INFO (MainThread) [sanic.root] Sanic Extensions:
[2022-07-27 09:56:01 -0400] [1405161] [INFO]   > templating [jinja2==2.11.2]
2022-07-27 09:56:01 INFO (MainThread) [sanic.root]   > templating [jinja2==2.11.2]
[2022-07-27 09:56:01 -0400] [1405161] [INFO]   > http
2022-07-27 09:56:01 INFO (MainThread) [sanic.root]   > http
[2022-07-27 09:56:01 -0400] [1405161] [INFO]   > openapi [http://0.0.0.0:9170/docs]
2022-07-27 09:56:01 INFO (MainThread) [sanic.root]   > openapi [http://0.0.0.0:9170/docs]
[2022-07-27 09:56:01 -0400] [1405161] [INFO]   > injection [0 added]
2022-07-27 09:56:01 INFO (MainThread) [sanic.root]   > injection [0 added]
[2022-07-27 09:56:01 -0400] [1405161] [INFO] Starting worker [1405161]
2022-07-27 09:56:01 INFO (MainThread) [sanic.root] Starting worker [1405161]
2022-07-27 09:56:03 INFO (MainThread) [scheduler] RUN TASK "_after_startup"
2022-07-27 09:56:03 INFO (MainThread) [scheduler] END TASK "_after_startup"
2022-07-27 09:56:03 INFO (MainThread) [scheduler] STOP TASK "_after_startup"
2022-07-27 09:56:03 INFO (Thread-2) [systembridgebackend.server.Server] Test WebSocket connection before starting GUI. Attempt #1
2022-07-27 09:56:03 INFO (Thread-2) [systembridgeshared.websocket_client.WebSocketClient] Connecting to WebSocket: ws://localhost:9170/api/websocket
[2022-07-27 09:56:03 -0400] [1405161] [INFO] connection open
2022-07-27 09:56:03 INFO (MainThread) [sanic.root] connection open
2022-07-27 09:56:03 INFO (Thread-2) [systembridgeshared.websocket_client.WebSocketClient] Connected to WebSocket
2022-07-27 09:56:03 INFO (Thread-2) [systembridgeshared.websocket_client.WebSocketClient] Closing WebSocket connection
2022-07-27 09:56:03 INFO (MainThread) [systembridgebackend.server.websocket.WebSocketHandler] Unregistering data listener 54134567-c9c3-45cc-aaf0-1ef5326bc8fe
2022-07-27 09:56:03 INFO (MainThread) [systembridgebackend.modules.listeners.Listeners] Listener not found: 54134567-c9c3-45cc-aaf0-1ef5326bc8fe
2022-07-27 09:56:03 INFO (Thread-2) [systembridgebackend.server.Server] Starting GUI: ['/usr/bin/python3', '-m', 'systembridgegui', 'main']
2022-07-27 09:56:03 INFO (Thread-2) [systembridgebackend.server.Server] GUI started with PID: 1405237
2022-07-27 09:56:03 INFO (MainThread) [__main__.Main] System Bridge GUI 3.4.3: Startup
2022-07-27 09:56:03 INFO (MainThread) [__main__.Main] Main: Setup
2022-07-27 09:56:03 INFO (MainThread) [systembridgeshared.websocket_client.WebSocketClient] Connecting to WebSocket: ws://localhost:9170/api/websocket
[2022-07-27 09:56:03 -0400] [1405161] [INFO] connection open
2022-07-27 09:56:03 INFO (MainThread) [sanic.root] connection open
2022-07-27 09:56:03 INFO (MainThread) [systembridgeshared.websocket_client.WebSocketClient] Connected to WebSocket
2022-07-27 09:56:03 INFO (MainThread) [systembridgeshared.websocket_client.WebSocketClient] Closing WebSocket connection
2022-07-27 09:56:03 INFO (MainThread) [systembridgebackend.server.websocket.WebSocketHandler] Unregistering data listener 9cc02d83-6930-4a45-86ac-da1c47e78f4f
2022-07-27 09:56:03 INFO (MainThread) [systembridgebackend.modules.listeners.Listeners] Listener not found: 9cc02d83-6930-4a45-86ac-da1c47e78f4f
2022-07-27 09:56:03 INFO (MainThread) [systembridgegui.system_tray.SystemTray] Setup system tray
2022-07-27 09:56:11 INFO (MainThread) [scheduler] RUN TASK "_update_data"
2022-07-27 09:56:11 INFO (MainThread) [scheduler] END TASK "_update_data"
2022-07-27 09:56:11 INFO (MainThread) [scheduler] RUN TASK "_update_frequent_data"
2022-07-27 09:56:11 INFO (Thread-3) [systembridgebackend.modules.update.Update] Update data
2022-07-27 09:56:11 INFO (Thread-5) [systembridgebackend.modules.update.Update] Update frequent data
2022-07-27 09:56:11 INFO (MainThread) [scheduler] END TASK "_update_frequent_data"
2022-07-27 09:56:11 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO sensors (key, type, name, hardware_type, hardware_name, value, timestamp) VALUES ('temperatures_coretemp_critical', 'critical', 'critical', 'coretemp', 'coretemp', '100.0', 1658930171.1941192) ON CONFLICT(key) DO UPDATE SET value = '100.0', timestamp = 1658930171.1941192 WHERE key = 'temperatures_coretemp_critical'
cannot start a transaction within a transaction
2022-07-27 09:56:11 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop0_write_time', '0', 1658930171.301396) ON CONFLICT(key) DO UPDATE SET value = '0', timestamp = 1658930171.301396 WHERE key = 'io_counters_per_disk_loop0_write_time'
cannot start a transaction within a transaction
2022-07-27 09:56:11 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop0_busy_time', '12', 1658930171.323819) ON CONFLICT(key) DO UPDATE SET value = '12', timestamp = 1658930171.323819 WHERE key = 'io_counters_per_disk_loop0_busy_time'
cannot start a transaction within a transaction
2022-07-27 09:56:11 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop1_busy_time', '28', 1658930171.4405606) ON CONFLICT(key) DO UPDATE SET value = '28', timestamp = 1658930171.4405606 WHERE key = 'io_counters_per_disk_loop1_busy_time'
cannot commit - no transaction is active
2022-07-27 09:56:11 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Refresh data by module: sensors
2022-07-27 09:56:11 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Data changed for module: sensors
2022-07-27 09:56:11 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop2_write_time', '0', 1658930171.4894257) ON CONFLICT(key) DO UPDATE SET value = '0', timestamp = 1658930171.4894257 WHERE key = 'io_counters_per_disk_loop2_write_time'
cannot start a transaction within a transaction
2022-07-27 09:56:11 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: DELETE FROM display
cannot commit - no transaction is active
2022-07-27 09:56:11 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Refresh data by module: display
2022-07-27 09:56:11 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Data changed for module: display
2022-07-27 09:56:11 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Refresh data by module: gpu
2022-07-27 09:56:11 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Data changed for module: gpu
2022-07-27 09:56:11 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO cpu (key, value, timestamp) VALUES ('frequency_1_current', '3618.836', 1658930171.660032) ON CONFLICT(key) DO UPDATE SET value = '3618.836', timestamp = 1658930171.660032 WHERE key = 'frequency_1_current'
not an error
2022-07-27 09:56:11 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO cpu (key, value, timestamp) VALUES ('frequency_1_max', '3700.0', 1658930171.6775386) ON CONFLICT(key) DO UPDATE SET value = '3700.0', timestamp = 1658930171.6775386 WHERE key = 'frequency_1_max'
cannot start a transaction within a transaction
2022-07-27 09:56:11 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO cpu (key, value, timestamp) VALUES ('stats_ctx_switches', '1270951672', 1658930171.7770486) ON CONFLICT(key) DO UPDATE SET value = '1270951672', timestamp = 1658930171.7770486 WHERE key = 'stats_ctx_switches'
not an error
2022-07-27 09:56:11 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop5_read_merged_count', '0', 1658930171.8207452) ON CONFLICT(key) DO UPDATE SET value = '0', timestamp = 1658930171.8207452 WHERE key = 'io_counters_per_disk_loop5_read_merged_count'
cannot commit - no transaction is active
2022-07-27 09:56:11 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO cpu (key, value, timestamp) VALUES ('times_iowait', '337.28', 1658930171.8748832) ON CONFLICT(key) DO UPDATE SET value = '337.28', timestamp = 1658930171.8748832 WHERE key = 'times_iowait'
cannot start a transaction within a transaction
2022-07-27 09:56:12 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop13_busy_time', '36', 1658930172.9447753) ON CONFLICT(key) DO UPDATE SET value = '36', timestamp = 1658930172.9447753 WHERE key = 'io_counters_per_disk_loop13_busy_time'
cannot start a transaction within a transaction
2022-07-27 09:56:13 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop15_read_count', '185', 1658930173.0387805) ON CONFLICT(key) DO UPDATE SET value = '185', timestamp = 1658930173.0387805 WHERE key = 'io_counters_per_disk_loop15_read_count'
cannot commit - no transaction is active
2022-07-27 09:56:13 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop15_write_count', '0', 1658930173.0445237) ON CONFLICT(key) DO UPDATE SET value = '0', timestamp = 1658930173.0445237 WHERE key = 'io_counters_per_disk_loop15_write_count'
cannot start a transaction within a transaction
2022-07-27 09:56:13 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO cpu (key, value, timestamp) VALUES ('times_per_cpu_0_softirq', '16.9', 1658930173.1802962) ON CONFLICT(key) DO UPDATE SET value = '16.9', timestamp = 1658930173.1802962 WHERE key = 'times_per_cpu_0_softirq'
cannot start a transaction within a transaction
2022-07-27 09:56:13 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO cpu (key, value, timestamp) VALUES ('times_per_cpu_1_user', '14786.95', 1658930173.198995) ON CONFLICT(key) DO UPDATE SET value = '14786.95', timestamp = 1658930173.198995 WHERE key = 'times_per_cpu_1_user'
cannot start a transaction within a transaction
2022-07-27 09:56:13 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop17_read_bytes', '102245376', 1658930173.2743058) ON CONFLICT(key) DO UPDATE SET value = '102245376', timestamp = 1658930173.2743058 WHERE key = 'io_counters_per_disk_loop17_read_bytes'
not an error
2022-07-27 09:56:13 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop17_write_bytes', '0', 1658930173.2849126) ON CONFLICT(key) DO UPDATE SET value = '0', timestamp = 1658930173.2849126 WHERE key = 'io_counters_per_disk_loop17_write_bytes'
cannot start a transaction within a transaction
2022-07-27 09:56:13 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop17_write_time', '0', 1658930173.3288124) ON CONFLICT(key) DO UPDATE SET value = '0', timestamp = 1658930173.3288124 WHERE key = 'io_counters_per_disk_loop17_write_time'
cannot start a transaction within a transaction
2022-07-27 09:56:13 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_per_disk_loop17_busy_time', '7996', 1658930173.3610198) ON CONFLICT(key) DO UPDATE SET value = '7996', timestamp = 1658930173.3610198 WHERE key = 'io_counters_per_disk_loop17_busy_time'
not an error
2022-07-27 09:56:13 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO cpu (key, value, timestamp) VALUES ('times_per_cpu_3_user', '15257.63', 1658930173.3658628) ON CONFLICT(key) DO UPDATE SET value = '15257.63', timestamp = 1658930173.3658628 WHERE key = 'times_per_cpu_3_user'
cannot commit - no transaction is active
2022-07-27 09:56:14 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('partitions_/sys/fs/cgroup/cpuset_device', 'cgroup', 1658930174.5432997) ON CONFLICT(key) DO UPDATE SET value = 'cgroup', timestamp = 1658930174.5432997 WHERE key = 'partitions_/sys/fs/cgroup/cpuset_device'
cannot start a transaction within a transaction
2022-07-27 09:56:14 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('partitions_/sys/fs/cgroup/cpuset_fstype', 'cgroup', 1658930174.5583963) ON CONFLICT(key) DO UPDATE SET value = 'cgroup', timestamp = 1658930174.5583963 WHERE key = 'partitions_/sys/fs/cgroup/cpuset_fstype'
cannot start a transaction within a transaction
2022-07-27 09:56:14 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('partitions_/sys/fs/cgroup/cpu,cpuacct_opts', 'rw,nosuid,nodev,noexec,relatime,cpu,cpuacct', 1658930174.7327383) ON CONFLICT(key) DO UPDATE SET value = 'rw,nosuid,nodev,noexec,relatime,cpu,cpuacct', timestamp = 1658930174.7327383 WHERE key = 'partitions_/sys/fs/cgroup/cpu,cpuacct_opts'
cannot start a transaction within a transaction
2022-07-27 09:56:14 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('partitions_/sys/fs/cgroup/devices_fstype', 'cgroup', 1658930174.7845378) ON CONFLICT(key) DO UPDATE SET value = 'cgroup', timestamp = 1658930174.7845378 WHERE key = 'partitions_/sys/fs/cgroup/devices_fstype'
cannot start a transaction within a transaction
2022-07-27 09:56:16 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO memory (key, value, timestamp) VALUES ('swap_percent', '3.0', 1658930176.9716334) ON CONFLICT(key) DO UPDATE SET value = '3.0', timestamp = 1658930176.9716334 WHERE key = 'swap_percent'
not an error
2022-07-27 09:56:17 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO memory (key, value, timestamp) VALUES ('swap_sin', '1564672', 1658930176.9963915) ON CONFLICT(key) DO UPDATE SET value = '1564672', timestamp = 1658930176.9963915 WHERE key = 'swap_sin'
cannot start a transaction within a transaction
2022-07-27 09:56:17 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('usage_/sys/fs/cgroup/cpuset_free', '0', 1658930177.0699537) ON CONFLICT(key) DO UPDATE SET value = '0', timestamp = 1658930177.0699537 WHERE key = 'usage_/sys/fs/cgroup/cpuset_free'
cannot start a transaction within a transaction
2022-07-27 09:56:17 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO network (key, value, timestamp) VALUES ('io_counters_packets_sent', '10718094', 1658930177.2753503) ON CONFLICT(key) DO UPDATE SET value = '10718094', timestamp = 1658930177.2753503 WHERE key = 'io_counters_packets_sent'
cannot start a transaction within a transaction
2022-07-27 09:56:17 ERROR (Thread-5) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO network (key, value, timestamp) VALUES ('io_counters_dropin', '1566', 1658930177.2975435) ON CONFLICT(key) DO UPDATE SET value = '1566', timestamp = 1658930177.2975435 WHERE key = 'io_counters_dropin'
cannot start a transaction within a transaction
2022-07-27 09:56:17 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Refresh data by module: cpu
2022-07-27 09:56:17 ERROR (Thread-3) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('usage_/sys/fs/cgroup/cpu,cpuacct_total', '0', 1658930177.3095672) ON CONFLICT(key) DO UPDATE SET value = '0', timestamp = 1658930177.3095672 WHERE key = 'usage_/sys/fs/cgroup/cpu,cpuacct_total'
cannot start a transaction within a transaction
2022-07-27 09:56:17 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Data changed for module: cpu
2022-07-27 09:56:17 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Refresh data by module: memory
2022-07-27 09:56:17 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Data changed for module: memory
2022-07-27 09:56:17 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Refresh data by module: network
2022-07-27 09:56:17 INFO (Thread-5) [systembridgebackend.modules.listeners.Listeners] Data changed for module: network
2022-07-27 09:56:17 INFO (Thread-5) [systembridgebackend.modules.update.Update] Finished updating frequent data
2022-07-27 09:56:20 INFO (Thread-3) [systembridgebackend.modules.system.System] Get latest version from GitHub
2022-07-27 09:56:20 INFO (Thread-3) [systembridgebackend.modules.listeners.Listeners] Refresh data by module: battery
2022-07-27 09:56:20 INFO (Thread-3) [systembridgebackend.modules.listeners.Listeners] Data changed for module: battery
2022-07-27 09:56:20 INFO (Thread-3) [systembridgebackend.modules.listeners.Listeners] Refresh data by module: disk
2022-07-27 09:56:20 INFO (Thread-3) [systembridgebackend.modules.listeners.Listeners] Data changed for module: disk
2022-07-27 09:56:21 INFO (Thread-3) [systembridgebackend.modules.listeners.Listeners] Refresh data by module: system
2022-07-27 09:56:21 INFO (Thread-3) [systembridgebackend.modules.listeners.Listeners] Data changed for module: system
2022-07-27 09:56:21 INFO (Thread-3) [systembridgebackend.modules.update.Update] Finished updating data
^C[2022-07-27 09:56:24 -0400] [1405161] [INFO] Stopping worker [1405161]
2022-07-27 09:56:24 INFO (MainThread) [sanic.root] Stopping worker [1405161]
[2022-07-27 09:56:24 -0400] [1405161] [INFO] Server Stopped
2022-07-27 09:56:24 INFO (MainThread) [sanic.root] Server Stopped
^C^C^C^C^C
```

### Additional information

I'm not sure what I should be seeing, but no GUI window pops up even though the logs above report that it started:

`2022-07-27 09:56:03 INFO (Thread-2) [systembridgebackend.server.Server] GUI started with PID: 1405237`

**Comment by timmo001** [comment]
Looks like the child process for the gui keeps running when SIGTERM is called. When I run the application with `python3 -m systembridgebackend --no-gui`, the server stops and the terminal is returned
